kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/flightSearchEngine/searching/SearchingService.java

```diff
@@ -0,0 +1,51 @@
+package com.kodilla.good.patterns.challenges.flightSearchEngine.searching;
+
+import com.kodilla.good.patterns.challenges.flightSearchEngine.Db.FlightsDb;
+import com.kodilla.good.patterns.challenges.flightSearchEngine.infoRetrieving.Flight;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class SearchingService {
+
+    private final List<Flight> flights;
+
+    public SearchingService(FlightsDb flightsDb) {
+        this.flights = flightsDb.getFlights();
+    }
+
+    public List<Flight> findFlightsFromCity(String departure) {
+        return flights.stream()
+                .filter(c -> c.getDeparture().equals(departure))
+                .collect(toList());
+    }
+
+    public List<Flight> findFlightsToCity(String arrival) {
+        return flights.stream()
+                .filter(f -> f.getArrival().equals(arrival))
+                .collect(toList());
+    }
+
+    public List<Flight> findOneStopFlight(String departure, String arrival) {
```

Comment:
Klasa `Flight` zawiera wyłącznie `departure` i `arrival`. Nie da się przy jej pomocy przedstawić lotu z przesiadką. Widzę, że z tej metody zwracasz listę `Flight`, czyli poszczególne "segmenty" jednego złożonego lotu są odrębnymi obiektami w tej liście, należącymi do jednego wyniku wyszukiwania.
Dla kontrastu, w dwóch powyższych metodach jeden obiekt na liście to jeden wynik wyszukiwania.
To jest mocno niekonsekwentne podejście - klasa `Flight` nie ma precyzyjnie zdefiniowanej odpowiedzialności i jest używana w dwóch niekompatybilnych kontekstach.